### PR TITLE
Part 2 price event orchestration

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

This is the second story in our journey of separating the Charge price handling flow into its' own architecture and at the same time reducing the architecture needed to process such a request (when compared to a Charge Information request).

With this story, we want to build the orchestration needed to process a PriceCommandReceived event, which basically means the structure needed to confirm/reject, persist, notify and raise integration event.
This is expected to introduce a few new services, which are to be used in the field between the ChargePriceCommandReceiverEndpoint and the database.

## References

* #1410

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+
+namespace GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommands.Validation.InputValidation.ValidationRules
+{
+    public class MaximumPriceRule : IValidationRuleWithExtendedData
+    {
+        private const int PriceUpperBound = 1000000;
+        private readonly ChargePriceOperationDto _chargeOperationDto;
+
+        public MaximumPriceRule(ChargePriceOperationDto chargeOperationDto)
```
Comment (x-platformcoder) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs:26]:
```suggestion
        public MaximumPriceRule(ChargePriceOperationDto chargePriceOperationDto)
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommands.Validation.InputValidation.ValidationRules;
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+
+namespace GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommands.Validation.InputValidation.Factories
+{
+    public class ChargePriceOperationInputValidationRulesFactory : IInputValidationRulesFactory<ChargePriceOperationDto>
```
Comment (x-platformcoder) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs:23]: Mangler test

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Linq;
+using GreenEnergyHub.Charges.Domain.Charges;
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+
+namespace GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommands.Validation.InputValidation.ValidationRules
+{
+    public class MaximumPriceRule : IValidationRuleWithExtendedData
```
Comment (x-platformcoder) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs:21]: mangler test

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/ChargePriceCommandReceivedEventHandler.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using GreenEnergyHub.Charges.Application.Charges.Services;
+using GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommandReceivedEvents;
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+using Microsoft.Extensions.Logging;
+
+namespace GreenEnergyHub.Charges.Application.Charges.Handlers
+{
+    public class ChargePriceCommandReceivedEventHandler : IChargePriceCommandReceivedEventHandler
```
Comment (x-platformcoder) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/ChargePriceCommandReceivedEventHandler.cs:24]: mangler test

**Review by x-platformcoder — COMMENTED** [review]
der mangler lidt test hist og her på de nye klasser der er splittet ud  i chargeprice.. tjek selv codecov

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/IChargePriceCommandReceivedEventHandler.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+using System.Threading.Tasks;
+using GreenEnergyHub.Charges.Domain.Dtos.ChargePriceCommandReceivedEvents;
+
+namespace GreenEnergyHub.Charges.Application.Charges.Handlers
+{
+    /// <summary>
+    /// Delegates a charge depending on if it contains
+    /// Charge Prices or Charge information.
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/IChargePriceCommandReceivedEventHandler.cs]: done, also changed in charge information flow

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs (hunk 3 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs:21]: done

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        private const int PriceUpperBound = 1000000;
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/MaximumPriceRule.cs]: done

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/ChargePriceCommandReceivedEventHandler.cs (hunk 4 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Charges/Handlers/ChargePriceCommandReceivedEventHandler.cs:24]: added

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs:23]: added

**Review by x-platformcoder — APPROVED** [review]
LGTM

**Pull request merged**